### DISCUSSION TITLE
Where to put images so that they are exported with the HTML?

### DISCUSSION THREAD

**Discussion by idgraham** [zim-desktop-wiki/zim-desktop-wiki#2698] [Q&A]
Hallo, from Wales UK, and thank you for having me. I'm in my 8th decade, and try to keep acquiring new skills, but I'm generally more at ease with a GUI than a terminal. 
I discovered Zim just a few days ago, when I switched my laptop to Zorin 17.2, and I'm very impressed with it. One thing I hope to do is make fairly extensive articles for my children and heirs and successor accessible as HTMLs. I have now got that sussed to my own satisfaction as far as text is concerned. But I'm still struggling to include images in the exported HTML. With dim memories of having made websites in the past, I have put the images in the folder of the Notebook they appear in. On the page they are absolutely fine. But when I export to HTML all I find are what I suppose are placeholders.
Any help very gratefully received. 
Sincerely
Ian Graham

**Comment by bruvellu** [comment]
Dear Ian,

When you export a page to HTML, the image files that are attached to the page are exported to a separate folder.  For example:

```
├── Page_with_images.html
└── Page_with_images_files
    ├── 20240422_123440.jpg
    └── 20240430_155813.jpg
```

Would you mind checking if this folder exists and if there are any images are inside it?

If the folder does not exist, it probably means that the image files are not attached to specific pages. You mentioned that you put the images inside your notebook folder. Could you try adding one of these images to a test page with `Insert` > `Image...` and then checking the checkbox that says `Attach image first`? (at the bottom of the dialog, see image below)

This will copy your image file as an attachment to the test page. Now, when you export the page as HTML, it should create the folder I mentioned above and the image should show up.

![image](https://github.com/user-attachments/assets/6ecdb5cc-c45f-40f0-b405-9b2af466ac67)

Please let me know how it goes.

Best,
Bruno

**Reply by idgraham** [reply]
bruvellu - thank you for your response, which came to me as an email, and I responded to it as such. I've now come back to the Forum thinking I should make some additional comment to my own original query, and have found it all again.

For general information: I think your note about the _Attach image first_ box is possibly the most crucial thing. I now do that, and have had some success in exporting *.HTML files which will open with the image. 
I think there is a second factor, which is the browser used to present these files. 
Routinely I use Opera as my default browser, because I like it as an online portal. 
But I think it is not great at opening saved *.HTML files  - I've had issues not just with Zim-derived files, but with pages saved from online. 
I also have Chromium installed, and that has opened for me a trial export from Zim of a single page including an image.

I would still be grateful for some definitive advice about the placing of images intended to be inserted into Zim and then an exported *.HTML file; and also about the best placing of the intended export file. 
Sincerely
Ian G.

**Comment by jungle-boogie** [comment]
I'm having the same issues as Ian. 

In my notebook directory, i have a single index.txt and a single jpg image. When I go to `insert > image` and browse to the image, i am able to see the image added to the page in zim-wiki.

However, when I export the journal, the image is not there. However, if i go to `insert > attachments`, and do an export, the jpg will be copied over. Of course, the image is not embedded into the document - just a link to the image, which properly displays.

**Comment by jaap-karssenberg** [comment]
This is the expected behavior. Only attachments are included in the export.
Other images are linked but this may go wrong depending on the location and
the filesystem layout.

Op di 27 mei 2025, 21:10 schreef jungle-boogie ***@***.***>:
